### ISSUE TITLE
Make --sjdbOverhang mutable via command-line in download_references.sh

### ISSUE THREAD

**Issue by MattBashton** [suhrig/arriba#9]
The script you provide for downloading the references + annotations assumes a read length of 201bp.

Specifically the final stage of the script builds a STAR index with:

`STAR --runMode genomeGenerate --genomeDir STAR_index_${ASSEMBLY}_${ANNOTATION} --genomeFastaFiles "$ASSEMBLY.fa" --sjdbGTFfile "$ANNOTATION.gtf" --runThreadN "$THREADS" --sjdbOverhang 200`

However if we consult the [STAR documentation ](https://github.com/alexdobin/STAR/blob/master/doc/STARmanual.pdf) note that the `-sjdbOverhang` for index generation is defined as:

>  Ideally, this length should be equal to the ReadLength-1, where ReadLength is the length of the reads

Consequently would hardwiring this to a fixed value of 200 be unwise?   For instance in my case I have 76bp reads, so my optimum value of `-sjdbOverhang` would be 75 assuming the STAR recommendations apply?

**Comment by suhrig** [comment]
It does not harm to set `--sjdbOverhang` too big. It will only make alignment marginally slower, because it increases the search space a little. Here is a statement from the developer of STAR about a similar question: https://groups.google.com/forum/#!topic/rna-star/h9oh10UlvhI

Setting it too small may lead to alignments being missed. So it should not be smaller than the recommended value, albeit any value >100 is overkill regardless of the read length according to the developer of STAR (see thread above). So I think it is safe to leave it at 200. At best, I might consider increasing it to 250, because some MiSeq platforms produce reads of that length, which is the longest (commonly) used read length I am aware of.

I must say that I have noticed hardly any differences between 100 and 200 in terms of both, alignment quality and speed.

BTW, you are free to use your own STAR index, if you already have one. The download script is only meant as a convenience wrapper around what is essentially a three step procedure:

1. Download annotation in GTF format.
2. Download assembly in FastA format.
3. Build STAR index from (1) and (2).

I figured that most people interested in Arriba use STAR anyway, so they can skip the download step entirely.
